handcalcs/render.py

```diff
@@ -39,7 +39,10 @@
 def render(line, cell):
     # Run the cell
     with cell_capture:
-        _nms.shell.run_cell(cell)
+        exec_result = _nms.shell.run_cell(cell)
+
+    if not exec_result.success:
+        return None
```

Comment:
What happens here is for something like:
```py
In [1] : %%render
           y = x***2 + 1

  File "<ipython-input-7-868a96f2430d>", line 1
    y = x***2 + 1
           ^
SyntaxError: invalid syntax
```

The cell actually returns `None`, and doesn't raise an error really. The error that's printed out here is because of execution of `_nms.shell_run_cell(cell)`. Or do we need change something here?


BTW, any idea how does `capture_output` differ depending the different values of its arguments `stdout, stderr` and `display`?

Reply:
> The cell actually returns None, and doesn't raise an error really. The error that's printed out here is because of execution of _nms.shell_run_cell(cell). Or do we need change something here?

No changes needed. I think you have the best solution.

> BTW, any idea how does capture_output differ depending the different values of its arguments stdout, stderr and display?

I do not know the specifics yet. At first, I was looking on the IPython documentation for the code that would "make it work". Now, we are getting into more of the finer details and fine-tuning behaviour so I am interested to know more.

Running the test suite is done with `pytest`. Cloning the repo and using `poetry` to install (with `poetry install` in the cloned project directory) will make short work of installing all dependencies, including the dev dependencies. Then you can run `poetry run pytest` to run the tests. 

You can look in the `travis.yml` file and run the same process described there. Once complete, navigate to the project directory and just run `pytest`